src/Naderman/Composer/AWS/AwsClient.php

```diff
@@ -35,6 +36,16 @@ class AwsClient
     protected $io;
 
     /**
+     * @var S3Client
+     */
+    protected $client;
```

Comment:
I'm going to see if I can address some of the issues in this PR.

It seems like @fdewinne keeps a reference to the client so that it doesn't have to get rebuilt for each download (e.g., packages.json + pack zip for every transitive dependency being loaded from S3).